### DISCUSSION TITLE
开源版的OpenAI ChatGPT Code interpreter实现

### DISCUSSION THREAD

**Discussion by 15849769265** [viopsa233/AI-testers-Forum#93] [技术交流]
用Flask运行Web服务，代码运行部分借助Jupyter kernel process，通过Kernel manager在Flask和Jupyter之间通信。UI部分是React做的，用户输入后，将输入转换成Prompt，借助GPT-4对任务分解并生成Python代码在Jupyter中执行。
![image](https://github.com/viopsa233/AI-testers-Forum/assets/104847582/3858062f-eb4e-4646-a46b-83bd7a1668b9)
项目链接：https://github.com/ricklamers/gpt-code-ui

**Comment by istudylove** [comment]
不错，支持
